### PR TITLE
feat: jumpstart model package arn instance type variants

### PR DESCRIPTION
**Pull request by evakravi**
*Issue #, if available:*

*Description of changes:*
JumpStart will be launching models that have model packages (arns) specific to instance types. To support this, we are modifying our metadata to include these arns in the variants metadata object, and also the SDK code which consumes this metadata to use the variant logic. 

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/jumpstart/artifacts/model_packages.py [hunk 1]
```diff
@@ -29,6 +29,7 @@
 def _retrieve_model_package_arn(
     model_id: str,
     model_version: str,
+    instance_type: Optional[str],
```
Comment (whittech1) [inline src/sagemaker/jumpstart/artifacts/model_packages.py:32]: Don't we need to do the same for `_retrieve_model_package_model_artifact_s3_uri()`?

File: tests/unit/sagemaker/jumpstart/test_artifacts.py [hunk 2]
```diff
@@ -129,3 +134,109 @@ def test_estimator_fit_kwargs(self, patched_get_model_specs):
...
+        )
+
+        assert (
+            _retrieve_model_package_arn(
+                region=region,
+                model_id=model_id,
+                scope=JumpStartScriptScope.INFERENCE,
+                model_version="*",
+                sagemaker_session=self.mock_session,
+                instance_type="nobodycares",
```
[the remark this answers is not in the document]
Comment (JGuinegagne) [reply tests/unit/sagemaker/jumpstart/test_artifacts.py:229]: wait till you receive the bill :)

**Review by JGuinegagne — APPROVED** [review]

File: tests/unit/sagemaker/jumpstart/test_artifacts.py [hunk 3]
```diff
@@ -129,3 +134,109 @@ def test_estimator_fit_kwargs(self, patched_get_model_specs):
...
+
+        with pytest.raises(ValueError):
+            _retrieve_model_package_arn(
+                region="cn-north-1",
+                model_id=model_id,
+                scope=JumpStartScriptScope.INFERENCE,
+                model_version="*",
+                sagemaker_session=self.mock_session,
+                instance_type="ml.p2.12xlarge",
+            )
```
[the remark this answers is not in the document]
Comment (evakravi) [reply tests/unit/sagemaker/jumpstart/test_artifacts.py:242]: i believe so

**Review by evakravi — COMMENTED** [review]
/bot run notebook-tests, pr

**Review by knikure — APPROVED** [review]

**Pull request merged**